src/llama_recipes/data/llama_guard/finetuning_data_formatter.py

```diff
@@ -0,0 +1,481 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# This software may be used and distributed according to the terms of the Llama Guard License Agreement.
+
+import copy
+import random
+from dataclasses import dataclass
+from enum import Enum
+from typing import Dict, List, Literal, Optional, Sequence
+
+
+@dataclass
+class Category:
+    name: str
+    description: str
+
+
+@dataclass
+class Guidelines:
```

Comment:
@MichaelTontchev This is a nice abstraction for handling prompt categories, maybe we can extract it and put it in the [prompt_format_utils](https://github.com/facebookresearch/llama-recipes/blob/main/src/llama_recipes/inference/prompt_format_utils.py#L14) in place of the SafetyCategories.

Reply:
cc @QingHu1227

Reply:
Sounds a good idea to me:)

Reply:
Will come in a later PR!